lib/msal-node/src/cache/NodeStorage.ts

```diff
@@ -218,13 +218,22 @@ export class NodeStorage extends CacheManager {
      * @param accountKey - lookup key to fetch cache type AccountEntity
      */
     getAccount(accountKey: string): AccountEntity | null {
-        const account = this.getItem(accountKey) as AccountEntity;
-        if (AccountEntity.isAccountEntity(account)) {
-            return account;
+        const accountEntity = this.getCachedAccountEntity(accountKey);
+        if (accountEntity && AccountEntity.isAccountEntity(accountEntity)) {
+            return this.updateOutdatedCachedAccount(accountKey, accountEntity);
         }
         return null;
     }
 
+    /**
+     * Reads account from cache, builds it into an account entity and returns it.
+     * @param accountKey
+     * @returns
+     */
+    getCachedAccountEntity(accountKey: string): AccountEntity | null {
+        return Object.assign(new AccountEntity(), this.getItem(accountKey));
```

Comment:
Will this throw if getItem returns undefined/null?

Reply:
It won't throw (Object.assign handles it gracefully) but this is actually a bug since it'll return an empty AccountEntity when it should be returning null. This is a great catch, thanks!